app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -117,6 +122,11 @@ def create
       respond_for_show(:resource => repository)
     end
 
+    api :GET, "/repositories/repository_types", N_("Show the available repository types")
+    def repository_types
+      render :json => Repository.creatable_repository_types.map { |repo_type| {:name => repo_type} }
```

Comment:
i would expect this to return all repository_types (not just creatable).  Why not return them all, but use a structure:

{:name => 'yum', :creatable => false}